02-express-tutorial/app.js

```diff
@@ -1 +1,63 @@
-console.log('Express Tutorial')
+const express = require("express");
+const { products } = require("./data");
+
+//invoke express
+const app = express();
+
+// setup static and middleware
+app.use(express.static("./public"));
+
+app.get("/api/v1/test", (req, res) => {
+  res.json({ message: "It worked!" });
+});
+
+app.get("/api/v1/products", (req, res) => {
+  res.json(products);
+});
+
+app.get("/api/v1/products/:productID", (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find((p) => p.id === idToFind);
+
+  if (!product) {
+    return res.status(404).send({ message: "That product was not found." });
+  }
+
+  return res.json(product);
+});
+//search 
+app.get('/api/v1/query', (req, res) => {
+  const { search, limit, regex, price } = req.query;
+  let sortedProducts = [...products];
```

Comment:
# Optional Improvement
[Naming]
The name `sortedProducts` is misleading as the products aren't actually sorted at this point. Or at least your handler here can't guarantee that if the dependency on `./data` were to change.
Using a name like that could give someone reading or maintaining the code at a later time the misconception that the variable is sorted and can be used as such. That could cause them to make mistakes or misunderstand how the code works.
... Naming is hard.

Reply:
Agree! 
Thanks a bunch; I appreciate all your knowledge.
I just duplicated the work of the instructor from the video.